laalaa/apps/advisers/tasks.py

```diff
@@ -82,11 +78,44 @@ def __init__(self, iterable=None, **kwargs):
         super(StrippedDict, self).__init__(iterable, **kwargs)
 
 
-class ImportProcess(Thread):
-    """
-    Loads/Updates data from xls spreadsheet
-    """
-
+class GeocoderTask(Task):
+
+    def __init__(self):
+        self.errors = []
+
+    def run(self, postcodes):
+        tot = len(postcodes)
+        for n, postcode in enumerate(postcodes):
+            pc = ' '.join(postcode[0].encode('utf-8').split())
+            err = None
+            try:
+                point = geocode(pc)
+            except geocoder.PostcodeNotFound:
+                err = 'Failed geocoding postcode: %s' % postcode
+                logging.warn(err)
+            except geocoder.GeocoderError as e:
+                err = 'Failed postcode: "%s" .Error connecting to ' \
+                      'geocoder: %s' % (pc, e)
+                logging.warn(err)
+
+            if err:
+                self.errors.append(err)
+
+            if point:
+                models.Location.objects.filter(
+                    postcode=pc
+                ).update(point=point)
```

Comment:
I don't think chaining the filter and the update here is a good idea for readability, it would be better (IMO) like this:
```python
locations = models.Location.objects.filter(postcode=pc)
locations.update(point=point)
```